src/mod_webdav.c

```diff
@@ -1753,37 +1757,45 @@ static handler_t mod_webdav_put(server *srv, connection *con, plugin_data *p, ha
 		 * Example: Content-Range: bytes 100-1037/1038 */
 
 		if (NULL != (b = http_header_request_get(con, HTTP_HEADER_OTHER, CONST_STR_LEN("Content-Range")))) {
-			const char *num = b->ptr;
-			off_t offset;
+			char *num = b->ptr;
+			off_t offset, end, length;
 			char *err = NULL;
 
-			if (0 != strncmp(num, "bytes ", 6)) {
-				con->http_status = 501; /* not implemented */
-
-				return HANDLER_FINISHED;
-			}
-
-			/* we only support <num>- ... */
+			/* parse range */
+			do {
+				if (0 != strncmp(num, "bytes ", 6)) break;
+				num += 6;
 
-			num += 6;
+				/* skip WS */
+				while (*num == ' ' || *num == '\t') num++;
 
-			/* skip WS */
-			while (*num == ' ' || *num == '\t') num++;
+				offset = strtoll(num, &err, 10);
+				if (num == err || *err != '-' || offset < 0) break;
+				num = err;
 
-			if (*num == '\0') {
-				con->http_status = 501; /* not implemented */
+				num++;
+				if (*num != '\0') {
+					end = strtoll(num, &err, 10);
+					if (num == err || *err != '/' || end < offset) break;
+					num = err;
 
-				return HANDLER_FINISHED;
-			}
-
-			offset = strtoll(num, &err, 10);
+					num++;
+					if (*num == '*') {
+						length = -1;
+					} else {
+						length = strtoll(num, &err, 10);
+						if (num == err || (length > 0 && length <= end)) break;
+					}
+				}
+				goto parse_success;
```

Comment:
I am not a fan of goto.  Please replace the goto by setting err = NULL here on success, and error out below this block if (NULL != err)